SPARCore/GADArticle.m

```diff
@@ -1,7 +1,95 @@
 #import "GADArticle.h"
+#import "GADRemoteModel.h"
 
 @implementation GADArticle
 
+static NSString *const API_ARTICLE_ID = @"id";
+static NSString *const API_AUTHORS = @"authors";
+static NSString *const API_BRIEF = @"brief";
+static NSString *const API_CONTENT = @"content";
+static NSString *const API_DATE_PUBLISHED = @"datePublished";
+static NSString *const API_HEADER_IMAGE = @"headerImage";
+static NSString *const API_PUBLICATION_ID = @"publication";
+static NSString *const API_TITLE = @"title";
+
+static NSString *const API_PUBLICATION_PATH = @"publications";
+static NSString *const API_ARTICLE_PATH = @"articles";
+
++ (void) articlesFromPublication: (NSString *)publicationId
+                                        completionHandler:(void(^_Nonnull)(NSArray<GADArticle *>
+                                                                           *_Nullable articles,
+                                                            NSError *_Nullable error))completion {
+    NSURL *queryURL = [GADArticle urlForArticlesFromPublication:publicationId];
+    
+    [GADRemoteModel fetchModelsWithParams:queryURL
+                          queryParameters:@{}
+                         modelTransformer:^(NSData *jsonData) {
+                            return [GADArticle articlesFromJSON:jsonData];
+                         }
+                        completionHandler:completion];
+
+}
+
+
+- (void) fetchFulltextWithCompletion: (void(^_Nonnull)(GADRemoteModel * *_Nullable model,
+                                                NSError *_Nullable error))completion {
+    NSURL *queryURL = [GADArticle createURLWithArticle:self.articleId publication:self.publicationId];
+    
+    [GADRemoteModel fetchModelsWithParams:queryURL
+                          queryParameters:@{}
+                         modelTransformer:^(NSData *jsonData){
+                             return [GADArticle articlesFromJSON:jsonData];
+                         }
+                        completionHandler:^(NSArray <GADArticle *> *_Nullable model, NSError *_Nullable error){
+                            GADArticle *fullArticle = model[0];
+                            self.content = fullArticle.content;
+                            self.authors = fullArticle.authors;
+                            //completion(); --How should we call the completion handler here?
+                        }];
+    
+}
+
++ (NSArray <GADArticle *> *) articlesFromJSON: (NSData *)json {
+    
+    NSMutableArray <GADArticle *> *articles = [[NSMutableArray alloc] init];
+    NSError *JSONParsingError;
+    NSArray *jsonArray = [NSJSONSerialization JSONObjectWithData:json
+                                                         options:kNilOptions error:&JSONParsingError];
+    
+    for (NSDictionary *element in jsonArray) {
+        //Map fields of element to fields of article
+        GADArticle *article = [[GADArticle alloc] init];
```

Comment:
We could also make a function on `GADArticle` like `+ (instancetype)articleFromDictionary:(NSDictionary*)dict;` to make this more concise.